service/src/main/java/eu/europa/ec/fisheries/ers/service/bean/AssetModuleServiceBean.java

```diff
@@ -64,26 +80,64 @@
         }
         */
 
-        return assetList;
+        return new ArrayList<>();
     }
 
     /**
      * {@inheritDoc}
      */
     @Override
-    public List<String> getAssetGuids(Collection<VesselIdentifierEntity> vesselIdentifiers) throws ServiceException {
-        /*
-        String request;
-        try {
-            request = AssetModuleRequestMapper.createAssetListModuleRequest(createAssetListQuery(vesselIdentifiers));
-        } catch (AssetModelMapperException e) {
-            log.error("Error while mapping vesselIdentifiers to create AssetListQuery!", e);
-            throw new ServiceException(e.getMessage(), e.getCause());
+    public List<String> getAssetGuids(Collection<VesselIdentifierEntity> vesselIdentifiers) {
+        AssetQuery assetQuery = new AssetQuery();
+
+        Map<VesselTypeAssetQueryEnum, List<String>> identifierMap = new HashMap<>();
+
+        for (VesselIdentifierEntity vesselIdentifier : vesselIdentifiers) {
+            VesselTypeAssetQueryEnum queryEnum = VesselTypeAssetQueryEnum.getVesselTypeAssetQueryEnum(vesselIdentifier.getVesselIdentifierSchemeId());
+            if (queryEnum != null && queryEnum.getConfigSearchField() != null && StringUtils.isNotEmpty(vesselIdentifier.getVesselIdentifierId())) {
+                List<String> queryEnumValues = identifierMap.get(queryEnum);
+                if (queryEnumValues == null) {
+                    queryEnumValues = new ArrayList<>();
+                    identifierMap.put(queryEnum, queryEnumValues);
+                }
+                queryEnumValues.add(vesselIdentifier.getVesselIdentifierId());
+            }
         }
-        return getGuidsFromAssets(request);
 
-         */
-        return new ArrayList<>();
+        Set<VesselTypeAssetQueryEnum> vesselTypeAssetQueryEnums = identifierMap.keySet();
+        for (VesselTypeAssetQueryEnum vesselTypeAssetQueryEnum : vesselTypeAssetQueryEnums) {
+            switch (vesselTypeAssetQueryEnum) {
+                case CFR:
+                    assetQuery.setCfr(identifierMap.get(vesselTypeAssetQueryEnum));
```

Comment:
Do a "List<String> blabla = identifierMap.get(vesselTypeAssetQueryEnum)" once before the switch and then use the blabla variable, instead of inside each case.